gpbft/gpbft.go

```diff
@@ -1320,26 +1332,19 @@ func (c *convergeState) FindMaxTicketProposal(table PowerTable) ConvergeValue {
 	// If the same ticket is used for two different values then either we get a decision on one of them
 	// only or we go to a new round. Eventually there is a round where the max ticket is held by a
 	// correct participant, who will not double vote.
-	var maxTicket *big.Int
-	var maxValue ConvergeValue
-
-	for key, value := range c.values {
-		for _, ticket := range c.tickets[key] {
-			senderPower, _ := table.Get(ticket.Sender)
-			ticketHash := blake2b.Sum256(ticket.Ticket)
-			ticketAsInt := new(big.Int).SetBytes(ticketHash[:])
-			weightedTicket := new(big.Int).Mul(ticketAsInt, big.NewInt(int64(senderPower)))
-			if maxTicket == nil || weightedTicket.Cmp(maxTicket) > 0 {
-				maxTicket = weightedTicket
-				maxValue = value
-			}
+	var bestValue ConvergeValue
```

Comment:
We can simplify this logic by initializing this value to our own value. Then we can skip the `c.HasSelfValue` part.

Reply:
I removed the self-value and did some cleanup; please review the last commit.